src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -191,6 +183,16 @@ internal class PersistentStreamPullingAgent : SystemTarget, IPersistentStreamPul
                 // We already logged individual exceptions for individual calls to Shutdown. No need to log again.
             }
 
+            var unregisterTasks = new List<Task>();
```

Comment:
Should we add a timer!=null check to "public Task AddSubscriber(" and not add to this list if agent is being shutdown.  Otherwise pubsub could add subscribers after we've removed them.